app/controllers/resources_controller.rb

```diff
@@ -49,10 +49,16 @@ def check_permission
     privilege = params[:privilege]
     raise ArgumentError, "privilege" unless privilege
 
-    if assumed_role.allowed_to?(privilege, @resource)
-      head :no_content
-    else
-      head :not_found
-    end
+    result = assumed_role.allowed_to?(privilege, @resource)
+
+    Audit::Event::Check.new(
+      user: current_user,
+      resource: @resource,
+      privilege: privilege,
+      role: assumed_role,
+      success: result
+    ).log_to Audit.logger
+
+    head result ? :no_content : :not_found
```

Comment:
Maybe it's just me, but I find statements like this really hard to parse. It always take me an extra look, and a moment of wondering, "do the precedence rules mean that this is going to behave correctly?" I'd have a much easier time with:

```
head(result ? : no_content : :not_found)
```